### ISSUE TITLE
d.js 202 rate limit

### ISSUE THREAD

**Issue by s9anus98a** [deedy5/duckduckgo_search#264]
> 2024-11-28 15:25:48,670 - INFO - response: https://duckduckgo.com/?q=korupsi+anak+DPR 200 18573
> 
> 2024-11-28 15:25:49,252 - INFO - response: https://links.duckduckgo.com/d.js?q=korupsi+anak+DPR&kl=wt-wt&l=wt-wt&p=&s=10&df=&vqd=4-246253216555363508857724047191336301838&bing_market=wt-WT&ex=-1 200 15468
> 
> 2024-11-28 15:25:49,340 - INFO - response: https://links.duckduckgo.com/d.js?q=korupsi+anak+DPR&kl=wt-wt&l=wt-wt&p=&s=0&df=&vqd=4-246253216555363508857724047191336301838&bing_market=wt-WT&ex=-1 200 12532
> 
> 2024-11-28 15:25:49,477 - INFO - response: https://duckduckgo.com/?q=hukuman+anak+DPR 200 18734
> 
> 2024-11-28 15:25:49,593 - INFO - response: https://links.duckduckgo.com/d.js?q=hukuman+anak+DPR&kl=wt-wt&l=wt-wt&p=&s=10&df=&vqd=4-211186452142239153826965944095287160930&bing_market=wt-WT&ex=-1 202 555
> 
> 2024-11-28 15:25:49,598 - INFO - response: https://links.duckduckgo.com/d.js?q=hukuman+anak+DPR&kl=wt-wt&l=wt-wt&p=&s=0&df=&vqd=4-211186452142239153826965944095287160930&bing_market=wt-WT&ex=-1 202 540
> 
> 2024-11-28 15:25:49,599 - WARNING - Search attempt 1 failed: https://links.duckduckgo.com/d.js?q=hukuman+anak+DPR&kl=wt-wt&l=wt-wt&p=&s=0&df=&vqd=4-211186452142239153826965944095287160930&bing_market=wt-WT&ex=-1 202 Ratelimit
> 
> 2024-11-28 15:25:50,599 - WARNING - Web search failed after all retries.

script:

```
def search_web(query, max_results=5, max_retries=1):
    for attempt in range(max_retries):
        try:
            with DDGS() as ddgs:
                results = list(ddgs.text(query, max_results=max_results))

            filtered_snippets = []
            for r in results:
                cleaned_content = clean_html_content(r['body'])
                if cleaned_content:
                    filtered_snippets.append(
                        f"**[{r['title']}]({r['href']})**\n\n{cleaned_content}"
                    )
            return filtered_snippets
        except Exception as e:
            logger.warning(f"Search attempt {attempt + 1} failed: {e}")
            time.sleep(2 ** attempt)
```

**Comment by deedy5** [comment]
For now, try backend=“api” or “lite”

**Comment by deedy5** [comment]
fixed in v6.4.1
